### DISCUSSION TITLE
Issue when batch editing entry with an existing relationship

### DISCUSSION THREAD

**Discussion by rithu1998** [directus/directus#5436] [Archive]
Hi,
I have a collection called course_classification that has three fields that are O2M relationships and one field that is a M2M relationship. When using the batch editing feature within the course_classification collection, I see that entries that have existing relationships within them send a null id in their payload making it seem like the relationship is getting deleted. 
I am not editing the previous relationship but rather adding a new relationship to the entries I am batch editing. I have added some examples below, where the parent id (course_classification_id) is 517 and the added M2M relationship id is 163 but the existing 161 relationship id is getting deleted (sending null payload) when adding 163. Not sure why the payload is sending a null course_classification id when we are just adding a relationship. This makes it seem like the existing relationship is getting deleted. Please let me know if you have any idea why this is occurring. Thanks!

<img width="566" alt="Screen Shot 2021-05-03 at 10 45 32 AM" src="https://user-images.githubusercontent.com/27563697/116912311-e52d6d80-abfc-11eb-9c0f-d2ec70ad508b.png">
<img width="567" alt="Screen Shot 2021-05-03 at 10 43 31 AM" src="https://user-images.githubusercontent.com/27563697/116912302-e363aa00-abfc-11eb-9a91-15904dc038c8.png">
<img width="563" alt="Screen Shot 2021-05-03 at 10 44 01 AM" src="https://user-images.githubusercontent.com/27563697/116912309-e494d700-abfc-11eb-9f5d-bbeb9069dc11.png">

**Comment by rithu1998** [comment]
@rijkvanzanten Should I post this in the issues section?

**Reply by rijkvanzanten** [reply]
No that's ok! I haven't had the time yet to read through this and investigate 🙂 I'll move it to an issue if need be 👍🏻

**Comment by rijkvanzanten** [comment]
If you only select one item in a m2m/o2m interface, that's the one item that's being saved in the relation with the parent. In "normal" items, it'll show and save the ones you had previously selected, though in batch-editing, you're overriding the whole value. Make sure you select all the items you want to have in the relationship when batch editing 👍🏻
